logging/null_tracer.go

```diff
@@ -0,0 +1,57 @@
+package logging
+
+import (
+	"context"
+	"net"
+	"time"
+)
+
+var (
+	// The NullTracer is a Tracer that does nothing.
+	// It is useful for embedding. Don't modify this variable!
+	NullTracer Tracer = &nullTracer{}
+	// The NullConnectionTracer is a ConnectionTracer that does nothing.
+	// It is useful for embedding. Don't modify this variable!
+	NullConnectionTracer ConnectionTracer = &nullConnectionTracer{}
```

Comment:
Design decision: Should `NullConnectionTracer` be a variable, or should it be the type?

Advantage variable: There's only a single `NullConnectionTracer` instance.
Advantage type: makes it really easy to embed, which is not possible when it's a variable.

Reply:
Switched it to a type. This makes embedding easier. I don't think having a single instance is actually a problem.